src/main/java/com/jcraft/jsch/JSch.java

```diff
@@ -507,7 +528,14 @@ public void addIdentity(String prvkey, String pubkey, byte[] passphrase) throws
    */
   public void addIdentity(String name, byte[] prvkey, byte[] pubkey, byte[] passphrase)
       throws JSchException {
-    Identity identity = IdentityFile.newInstance(name, prvkey, pubkey, instLogger);
+    String pubkeyFileContent = new String(pubkey, UTF_8);
```

Comment:
Is there a way we can detect if this is a cert without having to turn the entire pubkey into a String?